tests/main/refresh-mode-ignore-running/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: apps can request not to be tracked
+
+systems: [-ubuntu-14.04-*]
+
+prepare: |
+  snap set system experimental.refresh-app-awareness=true
+  snap pack test-snapd-refresh
+  snap install --dangerous ./test-snapd-refresh_1_all.snap
+
+restore: |
+  snap remove --purge test-snapd-refresh
+  rm -f test-snapd-refresh_1.all.snap
+  snap unset system experimental.refresh-app-awareness
+
+execute: |
+  tests.cleanup defer rm -f stamp
+
+  trap 'touch stamp' EXIT
+
+  test-snapd-refresh.sh -c "while [ ! -e stamp ]; do sleep 1; done" &
+  not snap install --dangerous ./test-snapd-refresh_1_all.snap
+  touch stamp
+  wait
```

Comment:
I'm not sure that this is checking anything, at least according to the bash man page:

> If id is not given, all currently active child processes are waited for, and the return status is zero.

Here and below, we probably want to run "`wait -n`" instead. Though, to be 100% sure, I'd rather change the test a bit and do something more explicit, like (untested):
```sh
test-snapd-refresh.sh -c "while [ ! -e stamp ]; do sleep 1; touch still_alive; done" &
not snap install --dangerous ./test-snapd-refresh_1_all.snap
touch stamp
wait -n
if [ still_alive -ot stamp ]; then
    echo "Program unexpectedly killed while refreshing";
    exit 1
fi
```
Though, now that I think of that, filesystem timestamps have a resolution of up to a second, so this test would actually never fail. :-( But maybe this gives you some ideas.

Also, it's not clear to me what these two blocks do. It would be nice to have a comment explaining what `test-snapd-refresh.sh` does, and what we are expecting with this code.

Reply:
My understanding is that the `wait` is there just to make sure that the app exits before the next install so it's not interfering with the app-awareness check. If it wasn't there, we'd have a race condition between the loop checking the stamp and the next install. I do agree on the comment/clarity part, so I'll open a follow-up to (at least) clarify what this is doing